<launch>
    <arg name="_drone_id"      default="0"/>
    <!-- 扩展运动基元间隔时间 -->
    <arg name="pre_step_" />
    <arg name="step_time_" />
    <arg name="used_DIS_"/>
    <arg name="used_adsm_"/>
    <arg name="used_static_path" default="true" />
    <arg name="traj_horizon"   default="9.0"/>

    <arg name="map_size_x_"    default="30.0"/>
    <arg name="map_size_y_"    default="8.0"/>
    <arg name="map_size_z_"    default="2.0"/>

    <arg name="_use_UserGoal"  default="false"/>
    <arg name="_goal_x"        default="10.0"/>
    <arg name="_goal_y"        default="0.0"/>
    <arg name="_goal_yaw"      default="0.0"/>

    <arg name="odometry_topic" default="/Odometry"/>
    <arg name="cloud_topic"    default="/velodyne_points_world"/>
    <arg name="_perception_GroundTruth"    default="true"/>

    <node pkg="traj_planner" type="globalFsm_by_adsm" name="globalFsm_by_adsm" output="screen">
        <!-- Remap Topic -->
        <remap from="/odometry"                 to="$(arg odometry_topic)"/>
        <remap from="/waypoints"                  to="/move_base_simple/goal"/>
        <remap from="/vis_dense_kino_traj"        to="/vis_kino_traj"/>
        <remap from="/sdf_map/odom"               to="$(arg odometry_topic)"/>
        <remap from="/sdf_map/cloud"              to="$(arg cloud_topic)"/>

        <!-- for obs_manager -->
        <!-- 障碍物轨迹预测值话题 -->
        <remap from="/trajs_predicted"  to="/obstacle_prediction_node/obstacle_prediction/trajs_predicted"/>
        <!--障碍物轨迹真值话题-->
        <remap from="/trajs"            to="/Dyn_Obs_trajs"/>        
        <!-- 是否使用障碍物轨迹真值 -->
        <param name="obs_manager/use_GroundTruth" value="$(arg _perception_GroundTruth)"/>
        <param name="obs_manager/pre_step" value="$(arg pre_step_)" />
        <param name="obs_manager/step_time" value="$(arg step_time_)" />

        <param name="use_presetGoal"        value="$(arg _use_UserGoal)"/>
        <param name="goal_x"                value="$(arg _goal_x)"/>
        <param name="goal_y"                value="$(arg _goal_y)"/>
        <param name="goal_yaw"              value="$(arg _goal_yaw)"/>

        <!-- 控制点采样间隔 -->
        <param name="manager/init_traj_sample_time"   value="0.4" type="double"/>
        <!-- for FSM -->
        <param name="fsm/thresh_replan"     value="1.5" type="double"/>
        <param name="fsm/thresh_no_replan"  value="1.0" type="double"/>
        <param name="fsm/static_path"       value="$(arg used_static_path)" type="bool"/>

        <!-- kinodynamic path searching -->
        <param name="search/max_tau"            value="0.5" type="double"/>
        <param name="search/init_max_tau"       value="1.0" type="double"/>
        <param name="search/max_vel"            value="1.5" type="double"/>
        <param name="search/max_acc"            value="2.0" type="double"/>
        <param name="search/omega_max"          value="0.4" type="double"/>
        <param name="search/w_time"             value="10.0" type="double"/>
        <param name="search/horizon"            value="$(arg traj_horizon)"  type="double"/>
        <param name="search/lambda_heu"         value="10000.0" type="double"/>
        <param name="search/resolution_astar"   value="0.1" type="double"/>
        <param name="search/time_resolution"    value="0.8" type="double"/>
        <param name="search/margin"             value="0.2" type="double"/>
        <param name="search/allocate_num"       value="10000" type="int"/>
        <param name="search/check_num"          value="5" type="int"/>
        <!-- 角速度的限制 -->
        <param name="search/is_consider_omega"  value="false" type="bool"/>
        <!-- is_used_VO和is_used_DIS都是false时，路径搜索不进行动态障碍物碰撞检测 安全约束的设置 -->
        <param name="search/is_used_VO"         value="false" type="bool"/>
        <param name="search/is_used_DIS"        value="$(arg used_DIS_)" type="bool"/>
        <param name="search/is_used_adsm"       value="$(arg used_adsm_)" type="bool"/>

        <!-- 地图参数 -->
        <param name="sdf_map/resolution"            value="0.1" />
        <param name="sdf_map/map_size_x"            value="$(arg map_size_x_)" /> 
        <param name="sdf_map/map_size_y"            value="$(arg map_size_y_)" /> 
        <param name="sdf_map/map_size_z"            value="$(arg map_size_z_)" /> 
        <param name="sdf_map/local_update_range_x"  value="10.0" />
        <param name="sdf_map/local_update_range_y"  value="10.0" />
        <param name="sdf_map/local_update_range_z"  value="4.5" />
        <!-- 膨胀半径 -->
        <param name="sdf_map/obstacles_inflation"     value="0.1" /> 
        <param name="sdf_map/local_bound_inflate"     value="0.0"/>
        <param name="sdf_map/local_map_margin"        value="50"/>
        <param name="sdf_map/ground_height"           value="0.0"/>
        <!-- depth filter -->
        <param name="sdf_map/use_depth_filter" value="true"/>
        <param name="sdf_map/depth_filter_tolerance" value="0.15"/>
        <param name="sdf_map/depth_filter_maxdist"   value="5.0"/>
        <param name="sdf_map/depth_filter_mindist"   value="0.2"/>
        <param name="sdf_map/depth_filter_margin"    value="2"/>
        <param name="sdf_map/k_depth_scaling_factor" value="1000.0"/>
        <param name="sdf_map/skip_pixel" value="2"/>
        <!-- local fusion -->
        <param name="sdf_map/p_hit"  value="0.65"/>
        <param name="sdf_map/p_miss" value="0.35"/>
        <param name="sdf_map/p_min"  value="0.12"/>
        <param name="sdf_map/p_max"  value="0.90"/>
        <param name="sdf_map/p_occ"  value="0.80"/>
        <param name="sdf_map/min_ray_length" value="0.5"/>
        <param name="sdf_map/max_ray_length" value="10.0"/>

        <param name="sdf_map/esdf_slice_height" value="0.3"/>
        <param name="sdf_map/visualization_truncate_height"   value="2.49"/>
        <param name="sdf_map/virtual_ceil_height"   value="2.5"/>
        <param name="sdf_map/show_occ_time"  value="false"/>
        <param name="sdf_map/show_esdf_time" value="false"/>
        <param name="sdf_map/pose_type"     value="1"/>  
        <param name="sdf_map/frame_id"      value="world"/>        
    </node>
</launch>